### ISSUE TITLE
[Feature request] Attribute similar to `PhpIgnore` but more like "ignore if null"

### ISSUE THREAD

**Issue by mrhappyasthma** [StringEpsilon/PhpSerializerNET#33]
I'm looking to have an object that I'm using as a data model which will eventually be serialized. But I want to mark some fields as ignored, but not `PhpIgnore` (which skips it unconditionally) but something along the lines of "Ignore, but only if it's null".

For example:

```
public class MyClass {
    [PhpProperty("Foo")]
    public string Foo {get;set;};
    
    // ... Other stuff
}
```

If `Foo` is `null`, it will serialize along the lines of: `s:3:"Foo";N;`

I would like this output to be skipped entirely if it is null.

Maybe something like

```
public class MyClass {
    [PhpProperty("Foo")]
    [PhpIgnoreNull]   // Or similar
    public string Foo {get;set;};
    
    // ... Other stuff
}
```

**Comment by StringEpsilon** [comment]
I like this idea, but this might be better implemented as something more generalized.

Perhaps an abstract attribute that allows for custom handling of the raw serialization / deserialization data. I believe JSON.net has something along those lines.

```cs
public abstract SerializationFilter : Attribute {
     public string Serialize(object data)
}
```

And your specific implementation would be along the lines of:

```cs
public abstract PhpIgnoreNull : PhpSerializationFilter  {
     public string Serialize(string key, object value {
        if (data == null) {
            return null;
        } 
        return PhpSerializer.Serialize(key) + PhpSerializer.Serialize(value);
     }
}
```

Only pseudo-code to demonstrate the idea. Something similar could be done added to the de-serialization as well.

**Comment by mrhappyasthma** [comment]
I like the idea of a serialization filter. It gives much more flexibility.

**Comment by StringEpsilon** [comment]
I've been a bit distracted / busy with other stuff. But I fiddled around a bit just now and made this branch:

https://github.com/StringEpsilon/PhpSerializerNET/tree/SerializationFilter 

I'm still not sure what exactly the API should look like and how to make it flexible enough.  Ideas? Suggestions?

**Comment by mrhappyasthma** [comment]
Sorry for the slow reply. I think that's great. The test case covers my specific needs, but it'l be useful to be able to apply generic filters, so I think the `Serialize` method override should be quite flexible.

**Comment by mrhappyasthma** [comment]
Any updates on this? Happy to give anything a review if you wanna bounce some ideas off someone.

**Comment by StringEpsilon** [comment]
No, not really. Sorry about that. I just haven't gotten around to much programming aside from some light refactoring in recent weeks.
